bson/src/test/unit/org/bson/codecs/pojo/PojoCustomTest.java

```diff
@@ -387,4 +389,16 @@ public void testInvalidGetterAndSetterModelEncoding() {
     public void testInvalidGetterAndSetterModelDecoding() {
         decodingShouldFail(getCodec(InvalidGetterAndSetterModel.class), "{'integerField': 42, 'stringField': 'myString'}");
     }
+
+    @Test(expected = CodecConfigurationException.class)
+    public void testImmutableFinalMapAndCollectionsModel() {
```

Comment:
This proves the asymmetry when dealing with immutable collections / maps and shows that the propertyAccessor doesn't leak an error when trying to mutate.